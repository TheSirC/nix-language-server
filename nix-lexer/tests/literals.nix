_
_leading_underscore
FooBar123
fooBar123
foo_bar_123
foo-bar-123
let
in
assert

123
00001
.0
.0e4
.0E-4
.0E+4
123.
123.45
123.e45
123.E45
123.E+45
123.E-45
123.45e67

./.
./foo
./foo/bar_baz-quux-123
/lost+found
/.
~/.
~/foo/bar
/intentionally/invalid/

<nixpkgs>
<nix/foo_bar-baz+quux.nix>

https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz

""
" foo ${bar baz} \" \${quux} "
'' foo ${bar baz} ''' ''${quux} ''
