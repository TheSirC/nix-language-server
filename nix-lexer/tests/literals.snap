Kind: Ident, Span: [0, 1), Text: "_"
Kind: Whitespace, Span: [1, 2), Text: "\n"
Kind: Ident, Span: [2, 21), Text: "_leading_underscore"
Kind: Whitespace, Span: [21, 22), Text: "\n"
Kind: Ident, Span: [22, 31), Text: "FooBar123"
Kind: Whitespace, Span: [31, 32), Text: "\n"
Kind: Ident, Span: [32, 41), Text: "fooBar123"
Kind: Whitespace, Span: [41, 42), Text: "\n"
Kind: Ident, Span: [42, 53), Text: "foo_bar_123"
Kind: Whitespace, Span: [53, 54), Text: "\n"
Kind: Ident, Span: [54, 65), Text: "foo-bar-123"
Kind: Whitespace, Span: [65, 66), Text: "\n"
Kind: Ident, Span: [66, 69), Text: "let"
Kind: Whitespace, Span: [69, 70), Text: "\n"
Kind: Ident, Span: [70, 72), Text: "in"
Kind: Whitespace, Span: [72, 73), Text: "\n"
Kind: Ident, Span: [73, 79), Text: "assert"
Kind: Whitespace, Span: [79, 81), Text: "\n\n"
Kind: Literal { kind: Integer }, Span: [81, 84), Text: "123"
Kind: Whitespace, Span: [84, 85), Text: "\n"
Kind: Literal { kind: Integer }, Span: [85, 90), Text: "00001"
Kind: Whitespace, Span: [90, 91), Text: "\n"
Kind: Literal { kind: Float }, Span: [91, 93), Text: ".0"
Kind: Whitespace, Span: [93, 94), Text: "\n"
Kind: Literal { kind: Float }, Span: [94, 98), Text: ".0e4"
Kind: Whitespace, Span: [98, 99), Text: "\n"
Kind: Literal { kind: Float }, Span: [99, 104), Text: ".0E-4"
Kind: Whitespace, Span: [104, 105), Text: "\n"
Kind: Literal { kind: Float }, Span: [105, 110), Text: ".0E+4"
Kind: Whitespace, Span: [110, 111), Text: "\n"
Kind: Literal { kind: Float }, Span: [111, 115), Text: "123."
Kind: Whitespace, Span: [115, 116), Text: "\n"
Kind: Literal { kind: Float }, Span: [116, 122), Text: "123.45"
Kind: Whitespace, Span: [122, 123), Text: "\n"
Kind: Literal { kind: Float }, Span: [123, 130), Text: "123.e45"
Kind: Whitespace, Span: [130, 131), Text: "\n"
Kind: Literal { kind: Float }, Span: [131, 138), Text: "123.E45"
Kind: Whitespace, Span: [138, 139), Text: "\n"
Kind: Literal { kind: Float }, Span: [139, 147), Text: "123.E+45"
Kind: Whitespace, Span: [147, 148), Text: "\n"
Kind: Literal { kind: Float }, Span: [148, 156), Text: "123.E-45"
Kind: Whitespace, Span: [156, 157), Text: "\n"
Kind: Literal { kind: Float }, Span: [157, 166), Text: "123.45e67"
Kind: Whitespace, Span: [166, 168), Text: "\n\n"
Kind: Literal { kind: Path { trailing_slash: false } }, Span: [168, 171), Text: "./."
Kind: Whitespace, Span: [171, 172), Text: "\n"
Kind: Literal { kind: Path { trailing_slash: false } }, Span: [172, 177), Text: "./foo"
Kind: Whitespace, Span: [177, 178), Text: "\n"
Kind: Literal { kind: Path { trailing_slash: false } }, Span: [178, 200), Text: "./foo/bar_baz-quux-123"
Kind: Whitespace, Span: [200, 201), Text: "\n"
Kind: Literal { kind: Path { trailing_slash: false } }, Span: [201, 212), Text: "/lost+found"
Kind: Whitespace, Span: [212, 213), Text: "\n"
Kind: Literal { kind: Path { trailing_slash: false } }, Span: [213, 215), Text: "/."
Kind: Whitespace, Span: [215, 216), Text: "\n"
Kind: Literal { kind: Path { trailing_slash: false } }, Span: [216, 219), Text: "~/."
Kind: Whitespace, Span: [219, 220), Text: "\n"
Kind: Literal { kind: Path { trailing_slash: false } }, Span: [220, 229), Text: "~/foo/bar"
Kind: Whitespace, Span: [229, 230), Text: "\n"
Kind: Literal { kind: Path { trailing_slash: true } }, Span: [230, 253), Text: "/intentionally/invalid/"
Kind: Whitespace, Span: [253, 255), Text: "\n\n"
Kind: Literal { kind: PathTemplate { trailing_slash: false } }, Span: [255, 264), Text: "<nixpkgs>"
Kind: Whitespace, Span: [264, 265), Text: "\n"
Kind: Literal { kind: PathTemplate { trailing_slash: false } }, Span: [265, 291), Text: "<nix/foo_bar-baz+quux.nix>"
Kind: Whitespace, Span: [291, 292), Text: "\n"
Kind: Literal { kind: PathTemplate { trailing_slash: true } }, Span: [292, 316), Text: "<intentionally/invalid/>"
Kind: Whitespace, Span: [316, 318), Text: "\n\n"
Kind: Literal { kind: Uri }, Span: [318, 382), Text: "https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz"
Kind: Whitespace, Span: [382, 384), Text: "\n\n"
Kind: StringTerm { kind: Normal }, Span: [384, 385), Text: "\""
Kind: StringTerm { kind: Normal }, Span: [385, 386), Text: "\""
Kind: Whitespace, Span: [386, 387), Text: "\n"
Kind: StringTerm { kind: Normal }, Span: [387, 388), Text: "\""
Kind: StringLiteral, Span: [388, 393), Text: " foo "
Kind: Interpolate, Span: [393, 395), Text: "${"
Kind: Ident, Span: [395, 398), Text: "bar"
Kind: Whitespace, Span: [398, 399), Text: " "
Kind: Ident, Span: [399, 402), Text: "baz"
Kind: CloseBrace, Span: [402, 403), Text: "}"
Kind: StringLiteral, Span: [403, 416), Text: " \\\" \\${quux} "
Kind: StringTerm { kind: Normal }, Span: [416, 417), Text: "\""
Kind: Whitespace, Span: [417, 418), Text: "\n"
Kind: StringTerm { kind: Indented }, Span: [418, 420), Text: "\'\'"
Kind: StringLiteral, Span: [420, 425), Text: " foo "
Kind: Interpolate, Span: [425, 427), Text: "${"
Kind: Ident, Span: [427, 430), Text: "bar"
Kind: Whitespace, Span: [430, 431), Text: " "
Kind: Ident, Span: [431, 434), Text: "baz"
Kind: CloseBrace, Span: [434, 435), Text: "}"
Kind: StringLiteral, Span: [435, 450), Text: " \'\'\' \'\'${quux} "
Kind: StringTerm { kind: Indented }, Span: [450, 452), Text: "\'\'"
Kind: Whitespace, Span: [452, 453), Text: "\n"